src/UI/Root/Components/Header/EnvSelector/EnvironmentSelector.test.tsx

```diff
@@ -1,196 +1,283 @@
+import { FlatEnvironment } from "@/Core";
 import { AuthProvider, getStoreInstance, KeycloakAuthConfig, LocalConfig } from "@/Data";
-import { AuthTestWrapper, Environment, dependencies } from "@/Test";
-import { DependencyProvider, EnvironmentHandlerImpl } from "@/UI/Dependency";
+import { useGetEnvironments } from "@/Data/Managers/V2/Environment";
+import { useGetProjects } from "@/Data/Managers/V2/Project/GetProjects";
+import { AuthTestWrapper, Environment, MockedDependencyProvider, Project } from "@/Test";
+import { testClient } from "@/Test/Utils/react-query-setup";
 import { TestMemoryRouter } from "@/UI/Routing/TestMemoryRouter";
 import ErrorBoundary from "@/UI/Utils/ErrorBoundary";
 import { EnvSelectorWithData as EnvironmentSelector } from "./EnvSelectorWithData";
 import { EnvironmentSelectorItem } from "./EnvSelectorWrapper";
+const EnvSelectorWrapper = ({
+  onSelectEnvironment,
+  selectedEnvironment,
+}: {
+  onSelectEnvironment: (item: EnvironmentSelectorItem) => void;
+  selectedEnvironment?: FlatEnvironment;
+}) => {
+  const environments = useGetEnvironments().useOneTime(true);
+  const projects = useGetProjects().useOneTime();
+
+  return (
+    <EnvironmentSelector
+      environments={environments}
+      projects={projects}
+      onSelectEnvironment={onSelectEnvironment}
+      selectedEnvironment={selectedEnvironment}
+    />
+  );
+};
 
 const setup = (
   onSelectEnvironment: (item: EnvironmentSelectorItem) => void = () => {},
-  config: KeycloakAuthConfig | LocalConfig | undefined = undefined,
-  environments: FlatEnvironment[] = Environment.filterable
+  config: KeycloakAuthConfig | LocalConfig | undefined = undefined
 ) => {
-  const queryClient = new QueryClient();
-  const environmentHandler = EnvironmentHandlerImpl(useLocation, dependencies.routeManager);
   const store = getStoreInstance();
 
-  store.dispatch.environment.setEnvironments(RemoteData.success(environments));
-
-  store.dispatch.environment.setEnvironmentDetailsById({
-    id: "123",
-    value: RemoteData.success(Environment.filterable[0]),
-  });
-
   return (
     <ErrorBoundary>
       <TestMemoryRouter initialEntries={["/?env=123"]}>
-        <QueryClientProvider client={queryClient}>
-          <DependencyProvider dependencies={{ ...dependencies, environmentHandler }}>
-            <StoreProvider store={store}>
-              <AuthProvider config={config}>
-                <AuthTestWrapper dependencies={dependencies}>
-                  <EnvironmentSelector
-                    environments={RemoteData.success(environments)}
-                    onSelectEnvironment={onSelectEnvironment}
-                    selectedEnvironment={Environment.filterable[0]}
-                  />
-                </AuthTestWrapper>
-              </AuthProvider>
-            </StoreProvider>
-          </DependencyProvider>
+        <QueryClientProvider client={testClient}>
+          <StoreProvider store={store}>
+            <AuthProvider config={config}>
```

Comment:
is this meant to be outside the AuthTestWrapper? (this is out of scope I know)

Reply:
Hmm, I took a look on it and its wrapper on MockedDependencyProvider and it's in correct place